### PR TITLE
Create query resolvers for search filters

### LINKED ISSUE #1052: Bug Report: Fix non-working search filters in talawa-admin
**Describe the bug**
All the filters in the organization dashboard in `talawa-admin`are non-working as of now.
We need to create specific query resolvers for

- posts `postsByOrganizationConnection` 
- events `eventsByOrganizationConnection`
- users `usersConnection`

like in the case of organizations, we created `organizationsConnection` in talawa-api and then, update the query in `talawa-admin`.

**Expected behavior**
Get the filtered instances as we try to use search filter in `talawa-admin`.

**Actual behavior**
No effect of using search filter in `talawa-admin`.

**Screenshots**

**Additional details**
Related to [talawa-admin-issue-460](https://github.com/PalisadoesFoundation/talawa-admin/issues/460)

### PR DESCRIPTION
**Pull request by beingnoble03**
**What kind of change does this PR introduce?**
Create query resolver filters for search filters.

**Issue Number:**

Fixes #1052 

**Did you add tests for your changes?**
Partially added.

**Does this PR introduce a breaking change?**
Yes, [talawa-admin-issue-460](https://github.com/PalisadoesFoundation/talawa-admin/issues/460) should be simultaneously fixed.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing lint tests

**Comment by beingnoble03** [comment]
@palisadoes working on them. Also, once the tests are passed, Should I add tests for additional Query resolvers in this PR itself or create separate issues for them? Because adding tests in this PR for the added query resolvers will increase number of changes in this PR, making it difficult for review.

**Comment by palisadoes** [comment]
> @palisadoes working on them. Also, once the tests are passed, Should I add tests for additional Query resolvers in this PR itself or create separate issues for them? Because adding tests in this PR for the added query resolvers will increase number of changes in this PR, making it difficult for review.

I agree with your proposed approach

**Comment by beingnoble03** [comment]
Also, [talawa-admin-issue-460](https://github.com/PalisadoesFoundation/talawa-admin/issues/460) should be fixed before merging this.

**Comment by beingnoble03** [comment]
> > @palisadoes working on them. Also, once the tests are passed, Should I add tests for additional Query resolvers in this PR itself or create separate issues for them? Because adding tests in this PR for the added query resolvers will increase number of changes in this PR, making it difficult for review.
> 
> I agree with your proposed approach

Ok, Will create issues for the added Query resolvers once, it's merged.

**Comment by palisadoes** [comment]
1. @xoldyckk @Tajcore @Kevoniat Please review. 
1. This is a blocking PR

**Review by palisadoes — APPROVED** [review]

**Comment by xoldyckk** [comment]
@palisadoes The issues I mentioned weren't fixed yet.

**Comment by palisadoes** [comment]
I didn't see them in the workflow.

**Comment by xoldyckk** [comment]
@palisadoes you can't see them on your screen?
Screenshot_20230219_235702

**Comment by beingnoble03** [comment]
@xoldyckk I can't see them too.

**Comment by xoldyckk** [comment]
@palisadoes @beingnoble03 I forgot to submit my reviews. Lol sorry.

**Comment by xoldyckk** [comment]
@palisadoes either re-open the issue or create a new issue I guess.

**Comment by palisadoes** [comment]
@beingnoble03 Please  resubmit. I'm reopening the issue.

File: src/typeDefs/queries.ts [hunk 1]
```diff
@@ -25,10 +25,24 @@ export const queries = gql`
...
     getDonationById(id: ID!): Donation!
 
     getDonationByOrgId(orgId: ID!): [Donation]
 
+    getDonationByOrgIdConnection(
+      orgId: ID!
+      where: DonationWhereInput
+      first: Int
+      skip: Int
+    ): [Donation]!
```
[the remark this answers is not in the document]
Comment (beingnoble03) [reply src/typeDefs/queries.ts:44]: Fixed in #1075.

File: src/typeDefs/queries.ts [hunk 2]
```diff
@@ -25,10 +25,24 @@ export const queries = gql`
 
     eventsByOrganization(id: ID, orderBy: EventOrderByInput): [Event]
 
+    eventsByOrganizationConnection(
+      where: EventWhereInput
+      first: Int
+      skip: Int
+      orderBy: EventOrderByInput
+    ): [Event]!
```
[the remark this answers is not in the document]
Comment (beingnoble03) [reply src/typeDefs/queries.ts:33]: Fixed in #1075.

File: src/resolvers/Query/getDonationByOrgIdConnection.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+/**
+ * @name getDonationByOrgIdConnection a GraphQL Query
+ * @description returns list of donations as a transactions that matches the provided orgId property from database and all the query parameters
+ */
+export const getDonationByOrgIdConnection: QueryResolvers["getDonationByOrgIdConnection"] =
+  async (_parent, args) => {
+    const inputArg = getInputArg(args.where);
+
+    return await Donation.find({
```
[the remark this answers is not in the document]
Comment (beingnoble03) [reply src/resolvers/Query/getDonationByOrgIdConnection.ts:16]: Fixed in #1075.

File: src/resolvers/Query/eventsByOrganizationConnection.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    const events = await Event.find(inputArg as Interface_Event)
+      .sort(sort)
+      .limit(args.first!)
+      .skip(args.skip!)
+      .populate("creator", "-password")
+      .populate("tasks")
+      .populate("admins", "-password")
+      .lean();
+
+    events.forEach((event) => {
```
[the remark this answers is not in the document]
Comment (beingnoble03) [reply src/resolvers/Query/eventsByOrganizationConnection.ts:33]: I tried to filter subdocuments using mongoose query (using `aggregate()` function) but then I am unable to use `populate()` function and `lean()` function on aggregated document. So, I don't think we can use any mongoose query here to filter subdocs. For a similar use case, we used `filter()` in `eventsByOrganization.ts`
```javascript
    events.forEach((event) => {
      event.registrants = event.registrants.filter(
        (registrant: Interface_UserAttende) =>
          registrant.status === STATUS_ACTIVE
      );
    });
```

**Pull request merged**